src/react/index.ts

```diff
@@ -12,19 +12,19 @@ import autoAnimate, {
  */
 export function useAutoAnimate<T extends Element>(
   options: Partial<AutoAnimateOptions> | AutoAnimationPlugin = {}
-): [RefObject<T>, (enabled: boolean) => void] {
-  const element = useRef<T>(null)
+): [RefCallback<T>, (enabled: boolean) => void] {
   const [controller, setController] = useState<
     AnimationController | undefined
   >()
+  const element = useCallback((node: T) => {
+    if (node instanceof HTMLElement) {
+      setController(autoAnimate(node, options))
+    }
```

Comment:
```suggestion
    } else {
      setController(undefined);
    }
```

as otherwise the controller will never be garbage collected when the element is unmounted